Nix improvements

chris-kruining opened the pull request:
This PR contains:

- support for both db types
- update docs
- refactor flake structure to flake-parts (use of flake-utils is discouraged)
- add nix checks (tests, run with `nix flake check`)

chris-kruining committed:
feat: implement more postgres details

Will do something about the package compilation later on

chris-kruining committed:
feat: extract version from exs file

chris-kruining committed:
feat: implement flake with better default option values, checks(WIP),
and initial start for postgres

chris-kruining committed:
feat: convert package to multiple in order to add an postgres build

chris-kruining committed:
fix: remove symlink, windows and linux are imcompatible with symlinks

chris-kruining committed:
feat: intial implementation of url parser

chris-kruining committed:
fix: cast port to string

chris-kruining committed:
fix: systemd `ExecStartPre`

chris-kruining committed:
fix: remove leading slash from url path

chris-kruining committed:
fix: sqlite database path

chris-kruining committed:
feat: add sslmode for postgres connection

chris-kruining committed:
chore: refactor shell deps

chris-kruining committed:
fix: correct case mismatch in test

chris-kruining committed:
fix: conditionally set ssl option

I hope this is what is needed. Not an Elixir dev.

arsfeld commented on nix/modules/flake-module.nix:
```diff
@@ -0,0 +1,456 @@
+            DATABASE_PORT = toString url.port;
+            DATABASE_NAME = (url.path or "/mydia") |> lib.substring 1 (-1) |> lib.splitString "/" |> lib.head;
+            DATABASE_USER = url.user or "mydia";
+            DATABASE_PASSWORD = url.password or "mydia";
```
You could use the same approach you used for `secretKeyBaseFile` and add some conditionals to LoadCredentials. I found an example in nixpkgs using this pattern: https://github.com/NixOS/nixpkgs/blob/8429e36770290990df3b178fffccdc037145ac2b/nixos/modules/services/web-apps/photoprism.nix#L125

It's a bit more complex but it's more explicit, I tend to dislike generic list of strings, they're useful but I would add specific cases as much as possible.

chris-kruining committed:
feat: add warning for when a password is passed plaintext via the
postgres uri

Perhaps this should be an assertion instead, but I can imagine that for
development purposes it could be benificial to allow plaintext

chris-kruining committed:
feat: tighten password passing for postgres
